src/reactors/UnifiedReactor.sol

```diff
@@ -0,0 +1,85 @@
+import {IAuctionResolver} from "../interfaces/IAuctionResolver.sol";
+import {IDCARegistry} from "../interfaces/IDCARegistry.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {CurrencyLibrary} from "../lib/CurrencyLibrary.sol";
+
+/// @notice Unified reactor that supports both AllowanceTransfer and SignatureTransfer
+/// with pluggable auction mechanisms
+contract UnifiedReactor is BaseReactor {
+    using Permit2Lib for ResolvedOrder;
+    using CurrencyLibrary for address;
+
+    /// @notice thrown when an auction resolver is not set
+    error EmptyAuctionResolver();
+    /// @notice thrown when an order's nonce has already been used
+    error InvalidNonce();
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function _resolve(SignedOrder calldata signedOrder)
+        internal
+        view
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        (address auctionResolver, bytes memory orderData) = abi.decode(signedOrder.order, (address, bytes));
+
+        if (auctionResolver == address(0)) {
+            revert EmptyAuctionResolver();
+        }
+
+        SignedOrder memory resolverOrder = SignedOrder({order: orderData, sig: signedOrder.sig});
+
+        IAuctionResolver resolver = IAuctionResolver(auctionResolver);
+
+        resolvedOrder = resolver.resolve(resolverOrder);
+        //try resolver.resolve(resolverOrder) returns (ResolvedOrder memory result) {
+        //    resolvedOrder = result;
+        //} catch (bytes memory reason) {
+        //    // Translate OrderAlreadyFilled to InvalidNonce for compatibility with base reactor tests
+        //    if (reason.length >= 4) {
+        //        bytes4 errorSelector = bytes4(reason);
+        //        // PriorityAuctionResolver.OrderAlreadyFilled.selector = 0x0700c4b8
+        //        if (errorSelector == 0x0700c4b8) {
+        //            revert InvalidNonce();
+        //        }
+        //    }
+        //    // Re-throw the original error
+        //    assembly {
+        //        revert(add(reason, 0x20), mload(reason))
+        //    }
+        //}
+    }
+
+    /// @inheritdoc BaseReactor
+    function _transferInputTokens(ResolvedOrder memory order, address to) internal override {
+        if (order.info.additionalValidationData.length > 0 && order.info.additionalValidationData[0] == 0x01) {
+            // Use AllowanceTransfer for DCA orders
+            IAllowanceTransfer.AllowanceTransferDetails memory details = IAllowanceTransfer.AllowanceTransferDetails({
+                from: order.info.swapper,
+                to: to,
+                amount: uint160(order.input.amount),
+                token: address(order.input.token)
+            });
+
+            IAllowanceTransfer(address(permit2)).transferFrom(details.from, details.to, details.amount, details.token);
```

Comment:
this is the root of my security concern:
- this now skips the `witness` check which guarantees onchain that the user actually approved the order details itself. now the reactor has no such validation at all and by this code path no authentication has been done at all (except maybe hidden internally in the resolver)

- secondly - full approving the reactor could be quite dangerous; in this case since `from` is just a details field - it seems possible to construct an order that pulls from someone else's hanging approval? I think this could be mitigated at various levels but is much more of a footgun than previously

Reply:
Right it relies on external validation contracts rather than transfer with witness